lib/src/license_detection/license.dart

```diff
@@ -0,0 +1,247 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:meta/meta.dart';
+import 'package:pana/src/license_detection/diff.dart';
+import 'package:pana/src/license_detection/token_matcher.dart';
+import 'package:pana/src/license_detection/tokenizer.dart';
+
+import 'crc32.dart';
+
+@sealed
+class License {
+  /// SPDX identifier of the license, is empty in case of unknown license.
+  final String identifier;
+
+  /// Original text from the license file.
+  final String content;
+
+  /// Normalized [Token]s created from the original text.
+  final List<Token> tokens;
+
+  /// Map from [Token.value] to their number of occurrences in this license.
+  final Map<String, int> occurrences;
+
+  License._(this.content, this.tokens, this.occurrences, this.identifier);
+
+  factory License.parse(String identifier, String content) {
+    final tokens = tokenize(content);
+    final table = generateFrequencyTable(tokens);
+    return License._(content, tokens, table, identifier);
+  }
+}
+
+String tokensNormalizedValue(Iterable<Token> tokens) {
+  var buffer = StringBuffer();
+
+  for (var token in tokens) {
+    buffer.write(token.value);
+    buffer.write(' ');
```

Comment:
Could be:
```dart
return tokens.map((t) => t.value).join(' ');
```

Reply:
If this function is only used two places maybe it's fine to just replace it with `tokens.map((t) => t.value).join(' ');`